jvm/src/main/kotlin/app/cash/security_sdk/Certificate.kt

```diff
@@ -0,0 +1,35 @@
+package app.cash.security_sdk
+
+import app.cash.security_sdk.internal.providers.TrifleContentVerifierProvider
+import app.cash.security_sdk.internal.signers.TrifleContentSigner
+import okio.ByteString
+import org.bouncycastle.cert.X509CertificateHolder
+import org.bouncycastle.cert.X509v3CertificateBuilder
+import java.math.BigInteger
+import java.time.Instant
+import java.time.Period
+import java.util.Date
+
+/*
+ * Class representing a Trifle certificate.
+ *
+ * @property certificate - bytearray encoding of an x.509 certificate. Subject to change.
+ */
+class Certificate internal constructor(internal val certificate: ByteArray) {
+  /**
+   * Serialize the Trifle Certificate to a ByteArray so clients can store it for inclusion in future
+   * signing operations or provide it as a root certificate to verify signed messages.
+   */
+  fun serialize(): ByteArray {
+    return certificate
+  }
+
+  companion object {
+    /**
+     * Create a Trifle Certificate from its binary representation.
+     */
+    fun deserialize(bytes: ByteArray): Certificate {
+      return Certificate(bytes)
```

Comment:
not particular sure but this isnt exactly deserializing the bytes right? I would have expected this to get fed through bouncy castle's x509 cert constructor and having some attributes be projected onto this class?

Reply:
right now the class is just a container of an x.509 cert, so the bytes are just an encoded x.509 cert.  We may eventually want to do more than that, but for now just storing the x.509 cert seemed reasonable since that's the only thing that is signed.